dali/ft/filecopy.cpp

```diff
@@ -592,6 +596,24 @@ FileSprayer::FileSprayer(IPropertyTree * _options, IPropertyTree * _progress, IR
     encryptKey.set(options->queryProp(ANencryptKey));
     decryptKey.set(options->queryProp(ANdecryptKey));
 
+    fileUmask = -1;
+    if (options->hasProp(ANumask))
+    {
+        StringBuffer umaskStr;
+        options->getProp(ANumask, umaskStr);
+        errno = 0;
+        fileUmask = (int)strtol(umaskStr.str(), NULL, 8);
+        if (errno)
```

Comment:
The implementation may also set errno to EINVAL in case no conversion was performed (no digits seen, and 0 returned).
I will add code to check for 0 return val and endptr val to help identify a failed conversion.